### PR TITLE
feat: Add listen timeouts to iroha cli

### PR DESCRIPTION
**Pull request by aoyako**
Resolves https://github.com/hyperledger-iroha/iroha/issues/2941

Now iroha-cli can take argument `--timeout=<seconds>` for listen commands. The client will exit when no message is received during this interval.

---

### Review notes (optional)

- For complex PRs, try to provide some information on how to approach the review more effectively.
- For example, is there a natural order in which the affected files should be reviewed?

### Checklist

- [ ] I've read [`CONTRIBUTING.md`](../CONTRIBUTING.md).
- [ ] (optional) I've written unit tests for the code changes.
- [ ] All review comments have been resolved.
- [ ] All CI checks pass.

### CODE REVIEW

File: crates/iroha_cli/Cargo.toml [hunk 1]
```diff
@@ -46,3 +46,6 @@ futures = { workspace = true }
 [build-dependencies]
 vergen = { version = "8.3.1", default-features = false }
 color-eyre = "0.6.3"
+
+[dev-dependencies]
+iroha_test_network = { workspace = true }
```
Comment (mversic) [inline crates/iroha_cli/Cargo.toml:51]: this is not ok. `iroha_test_network` is for writing integration tests, you can't have it as dependency to other crates

File: crates/iroha_cli/src/main.rs [hunk 2]
```diff
@@ -354,22 +404,46 @@ mod blocks {
     pub struct Args {
         /// Block height from which to start streaming blocks
         height: NonZeroU64,
+
+        /// Wait timeout in seconds
```
Comment (DCNick3) [inline crates/iroha_cli/src/main.rs:408]: IMO, specifying this in seconds might not be enough granularity. How about making this milliseconds?

Comment (DCNick3) [reply crates/iroha_cli/src/main.rs:395]: I think either way is ok

Comment (0x009922) [reply crates/iroha_cli/src/main.rs:400]: In `multisig` subcommand, `multisig::Register.transaction_ttl` uses `humantime::Duration`. I'd suggest reusing it here for consistency.

**Review by s8sato — DISMISSED** [review]

File: pytests/iroha_cli_tests/test/events/__init__.py
[the remark this answers is not in the document]
Comment (DCNick3) [reply pytests/iroha_cli_tests/test/events/__init__.py:1]: I think it's fine that it's empty, it makes python treat the directory differently somehow ("as a package"): https://docs.python.org/3/tutorial/modules.html#packages

File: pytests/iroha_cli_tests/pyproject.toml [hunk 3]
```diff
@@ -8,6 +8,7 @@ readme = "README.md"
 [tool.poetry.dependencies]
 python = "^3.11"
 faker = "*"
+cffi = ">=1.17.0"
```
Comment (DCNick3) [inline pytests/iroha_cli_tests/pyproject.toml:11]: Why did you need to add this package? Was it to force an update to newer version?

(If that's the case, `poetry update` would've done it, no need to add a transitive dependency into `pyproject.toml`)

Comment (DCNick3) [reply pytests/iroha_cli_tests/pyproject.toml:11]: Just running poetry update will also get you `1.17.1` and put this update in the lockfile. No need to put this dep into `pyproject.toml`

Comment (aoyako) [reply pytests/iroha_cli_tests/pyproject.toml]: It looks like this line, indeed, just updated the direct dependencies.

**Review by s8sato — APPROVED** [review]
